src/main-process/comfyInstallation.ts

```diff
@@ -15,64 +14,158 @@ type InstallState = Exclude<DesktopSettings['installState'], undefined>;
-   * @returns The validated installation state, along with a list of any issues detected.
+   * Validate the installation and add any results to {@link ComfyInstallation.validation}.
+   * @returns The validated installation state
+   * @throws When the YAML file is present but not readable (access denied, FS error, etc).
    */
   async validate(): Promise<InstallState> {
     log.info(`Validating installation. Recorded state: [${this.state}]`);
-
-    let { state } = this;
+    const validation: InstallValidation = {
+      inProgress: true,
+      installState: this.state,
+    };
+    this.validation = validation;
+    const update = () => this.onUpdate?.(validation);
+    update();
 
     // Upgraded from a version prior to 0.3.18
     // TODO: Validate more than just the existence of one file
-    if (!state && ComfyServerConfig.exists()) {
+    if (!validation.installState && ComfyServerConfig.exists()) {
       log.info('Found extra_models_config.yaml but no recorded state - assuming upgrade from <= 0.3.18');
-      state = 'upgraded';
+      validation.installState = 'upgraded';
+      update();
     }
 
     // Validate base path
     const basePath = await this.loadBasePath();
-    if (basePath === undefined || !(await pathAccessible(basePath))) {
-      log.warn('"base_path" is inaccessible or undefined.');
-      this.issues.add('invalidBasePath');
+    if (basePath && (await pathAccessible(basePath))) {
+      validation.basePath = 'OK';
+      update();
+
+      // TODO: Validate content of venv, etc.
+      const venv = new VirtualEnvironment(basePath, this.device);
+      if (await venv.exists()) {
+        validation.venvDirectory = 'OK';
+        update();
+
+        // Python interpreter
+        validation.pythonInterpreter = (await canExecute(venv.pythonInterpreterPath)) ? 'OK' : 'error';
+        if (validation.pythonInterpreter !== 'OK') log.warn('Python interpreter is missing or not executable.');
+        update();
+
+        // uv
+        if (await canExecute(venv.uvPath)) {
+          validation.uv = 'OK';
+          update();
+
+          // Python packages
+          try {
+            validation.pythonPackages = (await venv.hasRequirements()) ? 'OK' : 'error';
+            if (validation.pythonPackages !== 'OK') log.error('Virtual environment is incomplete.');
+          } catch (error) {
+            log.error('Failed to read venv packages.', error);
+            validation.pythonPackages = 'error';
+          }
+        } else {
+          log.warn('uv is missing or not executable.');
+          validation.uv = 'error';
+        }
+      } else {
+        log.warn('Virtual environment is missing.');
+        validation.venvDirectory = 'error';
+      }
+    } else {
+      log.error('"base_path" is inaccessible or undefined.');
+      validation.basePath = 'error';
+    }
+    update();
+
+    // Git
+    // TODO: Accurate cross-platform PATH search `git` executable
```

Comment:
Can you file an issue tracking this TODO, and add the issue URL here? The issue should explain what needs to be done in detail.